### PR TITLE
Nexus 23680: helm hosted implementation for new DBs 

### PR DESCRIPTION
**Pull request by igorudovika**
Nexus 23680: helm hosted implementation for new DBs

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+@Named
+public class HelmHostedFacetImpl
+    extends FacetSupport
+    implements HelmHostedFacet
+{
+  private HelmContentFacet helmContentFacet;
+
+  @Inject
+  public HelmHostedFacetImpl()
+  {
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java:34]: You can add no-op comment here.

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HostedHandlers.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.sonatype.repository.helm.internal.util.HelmPathUtils;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.http.HttpResponses.notFound;
+import static org.sonatype.nexus.repository.http.HttpResponses.ok;
+
+/**
+ * Helm Hosted Handlers
+ *
+ * @since 0.0.2
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HostedHandlers.java:33]: Is it an actual version?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java (hunk 1 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java]: Fixed 134e0fecc36375934a081e33e7ea5d90e2c92183

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HostedHandlers.java (hunk 2 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HostedHandlers.java]: Fixed 134e0fecc36375934a081e33e7ea5d90e2c92183

**Comment by Darkvar** [comment]
LGTM

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    return helmContentFacet.getAsset(path).orElse(null);
+  }
+
+  @Override
+  public void upload(
+      final String path,
+      final Payload payload,
+      final AssetKind assetKind) throws IOException
+  {
+    // TODO: Write policy selector
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java:51]: `HelmHostedWritePolicySelector` is not implemented for the new DB.

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+  @Inject
+  Provider<HelmHostedFacetImpl> hostedFacet
+
+  @Inject
+  Provider<HelmSecurityFacet> securityFacet
+
+  @Inject
+  Provider<ConfigurableViewFacet> viewFacet
+
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy:78]: Minor: redundant line

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  HighAvailabilitySupportChecker highAvailabilitySupportChecker
+
+  @Inject
+  HelmHostedRecipe(@Named(HostedType.NAME) final Type type, @Named(HelmFormat.NAME) final Format format) {
+    super(type, format)
+  }
+
+  @Override
+  boolean isFeatureEnabled() {
+    return highAvailabilitySupportChecker.isSupported(getFormat().getValue())
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy:117]: Do we really support high availability for the new DB? Maybe we should hardcode false here?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/HelmUploadHandler.java [hunk 6]
```diff
@@ -62,12 +62,12 @@
 
   private final HelmAttributeParser helmPackageParser;
 
-
   @Inject
-  public HelmUploadHandler(final ContentPermissionChecker contentPermissionChecker,
-                           final HelmAttributeParser helmPackageParser,
-                           @Named("simple") final VariableResolverAdapter variableResolverAdapter,
-                           final Set<UploadDefinitionExtension> uploadDefinitionExtensions)
+  public HelmUploadHandler(
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/HelmUploadHandler.java:66]: Minor: redundant refactoring. Put "VCS only" checkbox in the refactoring menu.

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java (hunk 3 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java]: Fixed 88bac16ecc93e6c7eb6f3fde288232ef8ea0468c

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy (hunk 4 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy:89]: Fixed 88bac16ecc93e6c7eb6f3fde288232ef8ea0468c

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy (hunk 5 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy]: Make sense, fixed 88bac16ecc93e6c7eb6f3fde288232ef8ea0468c

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/HelmUploadHandler.java (hunk 6 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/HelmUploadHandler.java:66]: fixed 88bac16ecc93e6c7eb6f3fde288232ef8ea0468c

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy (hunk 5 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy]: Unfortunately ^ does not work. False means "format/recipe is unavailable".  isFeatureEnabled() should return true always. Raw was implemented in such way. So I assuming that highAvailability flags should not impact on repo functionality as far clustering is not supported for new DBs.

**Pull request merged**